build-support/run-clang-tidy.py

```diff
@@ -311,9 +311,16 @@ def update_progress(current_file, num_files):
 
         # Fill the queue with files.
         for i, name in enumerate(files):
-            if file_name_re.search(name):
-                task_queue.put(name)
-            update_progress(i, len(files))
+            put_file = False
+            while not put_file:
+                try:
+                    #print("\n" + name)
+                    if file_name_re.search(name):
+                        task_queue.put(name, block=True, timeout=300)
+                        put_file = True
+                    update_progress(i, len(files))
+                except queue.Full:
+                    print('Still waiting to put files into clang-tidy queue.')
```

Comment:
It's not obvious why you need this change, maybe a comment or two explaining how this changes behavior and why that is important would be helpful.  Also remove the commented out print statement.